microsoftdynamicscrm/write.go

```diff
@@ -2,10 +2,40 @@ package microsoftdynamicscrm
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/amp-labs/connectors/common"
 )
 
+type writeMethod func(context.Context, string, any) (*common.JSONHTTPResponse, error)
+
 func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) {
-	panic("not implemented")
+	if len(config.ObjectName) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	url := c.getURL(config.ObjectName)
+
+	var write writeMethod
+	if len(config.RecordId) == 0 {
+		// writing to the entity without id means
+		// that we are extending 'List' resource
+		write = c.post
+	} else {
+		// only patch is supported for updating 'Single' resource
+		write = c.patch
+		// resource id is passed via brackets in OData spec
+		url = fmt.Sprintf("%s(%s)", url, config.RecordId)
+	}
+
+	// Neither Post nor Patch return any response data on successful completion
+	// Both complete with 204 NoContent
```

Comment:
Can you add a comment right above the definition of this Write function (line 12) that says that common.WriteResult will only have `Success` populated.

Reply:
@laurenzlong, done :heavy_check_mark: